core/src/main/java/org/apache/accumulo/core/client/impl/Namespaces.java

```diff
@@ -100,50 +106,150 @@ private static ZooCache getZooCache(Instance instance) {
     return namespaceMap;
   }
 
-  public static boolean exists(Instance instance, String namespaceId) {
+  public static boolean exists(Instance instance, Namespace.ID namespaceId) {
     ZooCache zc = getZooCache(instance);
     List<String> namespaceIds = zc.getChildren(ZooUtil.getRoot(instance) + Constants.ZNAMESPACES);
-    return namespaceIds.contains(namespaceId);
-  }
-
-  public static String getNamespaceId(Instance instance, String namespace) throws NamespaceNotFoundException {
-    String id = getNameToIdMap(instance).get(namespace);
-    if (id == null)
-      throw new NamespaceNotFoundException(null, namespace, "getNamespaceId() failed to find namespace");
-    return id;
+    return namespaceIds.contains(namespaceId.canonicalID());
   }
 
+  /**
+   * @deprecated Do not use - String for namespace ID is not type safe. Use {@link #getNamespaceName(Instance, Namespace.ID)}
+   */
+  @Deprecated
   public static String getNamespaceName(Instance instance, String namespaceId) throws NamespaceNotFoundException {
     String namespaceName = getIdToNameMap(instance).get(namespaceId);
     if (namespaceName == null)
       throw new NamespaceNotFoundException(namespaceId, null, "getNamespaceName() failed to find namespace");
     return namespaceName;
   }
 
+  /**
+   * @deprecated Do not use - Not type safe, ambiguous String-String map. Use {@link #getNameMap(Instance)}
+   */
+  @Deprecated
   public static SortedMap<String,String> getNameToIdMap(Instance instance) {
     return getMap(instance, true);
   }
 
+  /**
+   * @deprecated Do not use - Not type safe, ambiguous String-String map. Use {@link #getIdMap(Instance)}
+   */
+  @Deprecated
   public static SortedMap<String,String> getIdToNameMap(Instance instance) {
     return getMap(instance, false);
   }
 
-  public static List<String> getTableIds(Instance instance, String namespaceId) throws NamespaceNotFoundException {
+  public static List<Table.ID> getTableIds(Instance instance, Namespace.ID namespaceId) throws NamespaceNotFoundException {
     String namespace = getNamespaceName(instance, namespaceId);
-    List<String> names = new LinkedList<>();
-    for (Entry<String,String> nameToId : Tables.getNameToIdMap(instance).entrySet())
+    List<Table.ID> tableIds = new LinkedList<>();
+    for (Entry<String,Table.ID> nameToId : Tables.getNameMap(instance).entrySet())
       if (namespace.equals(Tables.qualify(nameToId.getKey()).getFirst()))
-        names.add(nameToId.getValue());
-    return names;
+        tableIds.add(nameToId.getValue());
+    return tableIds;
```

Comment:
I bet this could be turned into a clever one-liner with Java 8 lambdas:

```java
String namespace = getNamespaceName(instance, namespaceId);
return Tables.getNameMap(instance).entrySet().stream().filter(e -> namespace.equals(Tables.qualify(e.getKey()).getFirst())).collect(Collectors.toList());
```